File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        maxP = 0
        while l <= len(prices) - 1:
            r = l + 1
            while r <= len(prices) - 1:
                maxP = max(maxP, (prices[r] - prices[l]))
                r += 1
            l += 1
        return maxP

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int maxA=0;
        int n = heights.size();
        int left=0;
        int right=n-1;
        while(left<right) {
            if(heights[left]<heights[right]){
                int minH=min(heights[left], heights[right]);
                int newMax=minH*(right-left);
                maxA = max(maxA, newMax);
                left++;
            }
            else {
                int minH=min(heights[left], heights[right]);
                int newMax=minH*(right-left);
                maxA = max(maxA, newMax);
                right--;
            }
        }
        return maxA;
    }
};
